plugins/woocommerce/src/Admin/RemoteInboxNotifications/IsWooExpressRuleProcessor.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Rule processor that passes (or fails) when the site is on the eCommerce
+ * plan.
+ *
+ * @package WooCommerce\Admin\Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Rule processor that passes (or fails) when the site is on the Woo Express plan.
+ * You may optionally pass a plan name to target specific Woo Express plan.
+ */
+class IsWooExpressRuleProcessor implements RuleProcessorInterface {
+	/**
+	 * Passes (or fails) based on whether the site is a Woo Express plan.
+	 *
+	 * @param object $rule         The rule being processed by this rule processor.
+	 * @param object $stored_state Stored state.
+	 *
+	 * @return bool The result of the operation.
+	 */
+	public function process( $rule, $stored_state ) {
+		if ( ! function_exists( 'wc_calypso_bridge_is_woo_express_plan' ) ) {
+			return false === $rule->value;
+		}
+
+		// If the plan is undefined, only check if it's a Woo Express plan.
+		if ( ! isset( $rule->plan ) ) {
+			return wc_calypso_bridge_is_woo_express_plan() === $rule->value;
+		}
+
+		// If a plan name is defined, only evaluate the plan if we're on the Woo Express plan.
+		if ( wc_calypso_bridge_is_woo_express_plan() ) {
+			$fn = 'wc_calypso_bridge_is_woo_express_' . (string) $rule->plan . '_plan';
+			if ( function_exists( $fn ) ) {
+				return $fn() === $rule->value;
+			}
+
+			$fn = 'wc_calypso_bridge_is_ecommerce_' . (string) $rule->plan . '_plan';
```

Comment:
@manospsyx why do we need to check for `is_ecommerce` related plans, if we target `woo_express` ?

Reply:
Thanks @manospsyx  - I've also renamed `is_wooexpress` to `is_woo_express` for consistency

See commit https://github.com/woocommerce/woocommerce/pull/40513/commits/8df23888b86348b69531d63cf2efddd72603d24b

Reply:
I'll retest this tomorrow, once this [PR](https://github.com/Automattic/wc-calypso-bridge/pull/1352) is deployed on all Woo Express sites and merge it